src/core/mod.rs

```diff
@@ -77,6 +79,7 @@ pub struct BPF {
     pub(crate) perf_events: HashSet<PerfEvent>,
     pub(crate) perf_events_array: HashSet<PerfEventArray>,
     pub(crate) xdp: HashSet<XDP>,
+    pub sockets: HashMap<String, i32>,
```

Comment:
I'm confused about this. Why is the visibility `pub` and not `pub(crate)`? Our other collections here tend to be expressed directly as `HashSet<type>` - is there a reason this must be a `HashMap<String, i32>`?

Reply:
When reading the packets the user needs the socket filedescriptor. As such the socket_fd must be accessible from the outside. It could be also realized with special getter methods. I used a HashMap here because for each interface a seperate socket is needed (as far as i know). In this way the sockets can be distinguished based on their interface name.

Reply:
I don't like this field being public. It leaks implementation details.

Reply:
Additionally - can we have more than one socket per interface? If so, this datastructure is insufficient.